R/get_truth.R

```diff
@@ -221,4 +221,76 @@ preprocess_usafacts <- function (save_location="./data-truth/usafacts/"){
+#' 
+#' @param save_location character specifying the location of to save raw truth data.
+#' Default to "./data-truth/"
+#' 
+#' @export
+preprocess_jhu <- function (save_location="./data-truth/"){
+  # Location name
+  location_names <- covidData::fips_codes[, c("location", "location_name")]
+  
+  # National data
+  national_cases_dframes <- covidData::load_jhu_data(spatial_resolution = "national",
+                                                     temporal_resolution = "daily",
+                                                     measure = "cases")
+  national_deaths_dframes <- covidData::load_jhu_data(spatial_resolution = "national",
+                                                      temporal_resolution = "daily",
+                                                      measure = "deaths")
+
+  # State data
+  state_cases_dframes <- covidData::load_jhu_data(spatial_resolution = "state",
+                                                     temporal_resolution = "daily",
+                                                     measure = "cases")
+  state_deaths_dframes <- covidData::load_jhu_data(spatial_resolution = "state",
+                                                      temporal_resolution = "daily",
+                                                      measure = "deaths")
+  
+  # County data
+  county_cases_dframes <- covidData::load_jhu_data(spatial_resolution = "county",
+                                                     temporal_resolution = "daily",
+                                                     measure = "cases")
+  county_deaths_dframes <- covidData::load_jhu_data(spatial_resolution = "county",
+                                                      temporal_resolution = "daily",
+                                                      measure = "deaths")
+  
+  # Combine to get cases data
+  cases_dframes <- dplyr::bind_rows(national_cases_dframes, state_cases_dframes, county_cases_dframes)
+  cases_dframes <- dplyr::left_join(cases_dframes, location_names, by = "location")
+  cases_dframes$inc[cases_dframes$inc < 0] <- 0
+  cases_dframes<-cases_dframes[!(cases_dframes$location==11001),]
+  
+  # Combine to get death data
+  deaths_dframes <- dplyr::bind_rows(national_deaths_dframes, state_deaths_dframes, county_deaths_dframes)
+  deaths_dframes <- dplyr::left_join(deaths_dframes, location_names, by = "location")
+  deaths_dframes$inc[deaths_dframes$inc < 0] <- 0
+  deaths_dframes<-deaths_dframes[!(deaths_dframes$location==11001),]
+  
+  # Get cumulative deaths
+  cumulative_deaths <- deaths_dframes[, c("date", "location", "location_name", "cum")]
+  colnames(cumulative_deaths)[colnames(cumulative_deaths) == 'cum'] <- 'value'
+  cumulative_deaths <- cumulative_deaths[!rowSums(is.na(cumulative_deaths[c("location_name","value")])), ]
+  readr::write_csv(cumulative_deaths, path = paste0(save_location,"truth-Cumulative Deaths.csv"))
```

Comment:
There's nothing wrong with this, but I'd note that in R a more standard approach would be to use the function `file.path` instead of `paste0` here.  In that case, you would provide `save_location` without the trailing `/`, i.e. `save_location = "./data-truth"` in the function call.

Reply:
Thanks for the suggestion! I have updated the code.